python/Generators/Generators.py

```diff
@@ -12,59 +14,23 @@ class Generators:
     def __init__(self, settings):
         self.settings = settings
         self.generator_list = settings.gens()
-        self.sherpa = None
-        self.whizard = None
-        self.madgraph = None
-        self.babayaga = None
-        self.proc_info = None
 
     def set_process_info(self, proc_info):
         self.proc_info = proc_info
 
     def initialize_generators(self):
-        if "Sherpa" in self.generator_list:
-            if Sherpa is not None:
-                self.sherpa = Sherpa.Sherpa(self.proc_info, self.settings)
-                self.sherpa.write_file()
-                self.sherpa.write_key4hepfile()
-            else:
-                print("Sherpa module not found. Unable to initialize Sherpa.")
-        if "Whizard" in self.generator_list:
-            if Whizard is not None:
-                self.whizard = Whizard.Whizard(self.proc_info, self.settings)
-                self.whizard.write_file()
-                self.whizard.write_key4hepfile()
-            else:
-                print("Whizard module not found. Unable to initialize Whizard.")
-
-        if "Madgraph" in self.generator_list:
-            if Madgraph is not None:
-                self.madgraph = Madgraph.Madgraph(self.proc_info, self.settings)
-                self.madgraph.write_file()
-                self.madgraph.write_key4hepfile()
-            else:
-                print("Madgraph module not found. Unable to initialize Madgraph.")
-
-        if "Babayaga" in self.generator_list:
-            if Babayaga is not None:
-                self.babayaga = Babayaga.Babayaga(self.proc_info, self.settings)
-                self.babayaga.write_file()
-                self.babayaga.write_key4hepfile()
-            else:
-                print("Babayaga module not found. Unable to initialize Babayaga.")
-
-        if "KKMC" in self.generator_list:
-            if KKMC is not None:
-                self.kkmc = KKMC.KKMC(self.proc_info, self.settings)
-                self.kkmc.write_file()
-                self.kkmc.write_key4hepfile()
-            else:
-                print("KKMC module not found. Unable to initialize KKMC.")
 
-        if "Pythia" in self.generator_list:
-            if Pythia is not None:
-                self.pythia = Pythia.Pythia(self.proc_info, self.settings)
-                self.pythia.write_file()
-                self.pythia.write_key4hepfile()
-            else:
-                print("Pythia module not found. Unable to initialize Pythia.")
+        for generatorName in self.generator_list:
+            # get the module
+            try:
+                generator   = importlib.import_module(generatorName)
```

Comment:
```suggestion
                generator   = importlib.import_module(f".{generatorName,}" package=__package__)
```
Should technically preserve enough of the package context to make things work without changing the generator `.py` files.

Alternatively the following might also work:
```python
generator = importlib.import_module(f"Generators.{generatorName}")
```

Reply:
generator = importlib.import_module(f"Generators.{generatorName}")
does not seem to work, will check why and try the other version

Reply:
Ah sorry, that was my mistake. I didn't realize this happens inside the `Generators` package / module. In that case, I think only the other option works (i.e. the one with the `.` relative import, specifying the `package` dynamically as well)

Reply:
I was imprecise, with a more detailed try structure:
        for generatorName in self.generator_list:
            # get the module
            try:
                generator   = importlib.import_module(f"Generators.{generatorName}")
                # get the ClassObject
                try:
                    generatorClass = getattr(generator,generatorName)
                    # execute the object
                    try:
                        generatorObj = generatorClass(self.proc_info, self.settings)
                        #writing file
                        print("execution ok")
                        generatorObj.write_file()
                        #writing key4hep file
                        print("wrote file")
                        generatorObj.write_key4hepfile()
                    except:
                        print(f"Requested Generator: {generatorName} object could not be executed, output files not written for {self.proc_info.get('_proclabel')}")
                except:
                    print(f"Requested Generator: {generatorName} class could not be loaded {self.proc_info.get('_proclabel')}")
            except:
                print(f"Requested Generator: {generatorName} could not be configured for {self.proc_info.get('_proclabel')}")

I get:
Requested Generator: Sherpa object could not be executed, output files not written for e- e+ -> mu- mu+.....
That would mean that:
- import worked
- getting the class worked
- but the execution of the class failed?
(or could the problem be hidden somewhere else?

Reply:
If I change the import structure of Sherpa.py from:
`from .GeneratorBase import GeneratorBase
from . import SherpaProcDB
`
to 
`from .GeneratorBase import GeneratorBase
from .SherpaProcDB import SherpaProcDB
`
it seems to work better, will check with all gens and the CI

Reply:
It actually works for all the generators, but why is:
`from .MadgraphProcDB import MadgraphProcDB`
needed when
`from Particles import Particle as part`
seems to work?

Reply:
Most likely `Particles` is visible from the `PYTHONPATH` because the top level `python` is available. The full (absolute) import would be `Generators.MadgraphProcDB`.

The difference between 
```python
from . import SherpaProcDB`
```
and 
```python
from .SherpaProcDB import SherpaProcDB
```
is that for the first case you import the `SherpaProcDB` module from the current scope, whereas in the second case you import the `SherpaProcDB` object (class, function, variable) from the local module `SherpaProcDB`. Hence the following two things are equivalent:
```python
from . import SherpaProcDB
sherpa = SherpaProcDB.SherpaProcDB()
```
and
```python
from .SherpaProcDB import SherpaProcDB
sherpa = SherpaProcDB()
```